### PR TITLE
Bumps collector version to v0.29.0

### PR DESCRIPTION
**Pull request by Raul9595**
Fixes: #323

### CODE REVIEW

**Comment by jpkrohling** [comment]
Based on the title of this PR, I changed it to be a draft PR. Mark this as ready for review when you are happy with the state of it.

**Comment by Raul9595** [comment]
Hi @jpkrohling. Sorry about that. I was yet to add the changelog entry which is why it was mentioned as a draft PR. It is ready for review now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Changes by Version
 ==================
 
+0.29.0 (2021-06-25)
+-------------------
+* Bumped OpenTelemetry Collector to v0.29.0
```
Comment (jpkrohling) [inline CHANGELOG.md:6]: Could you add the following change to the changelog? 
https://github.com/open-telemetry/opentelemetry-operator/pull/313

And while you are at it, please add this to v0.28.0:
https://github.com/open-telemetry/opentelemetry-operator/pull/302
https://github.com/open-telemetry/opentelemetry-operator/pull/299
https://github.com/open-telemetry/opentelemetry-operator/pull/293

cc @VineethReddy02

Comment (Raul9595) [reply CHANGELOG.md:6]: Sure I'll add it

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**